File: 1378-CellsWithOddValuesInAMatrix/1378-CellsWithOddValuesInAMatrix.cpp
// Last updated: 18/10/2025, 09:03:04
class Solution {
public:
    int oddCells(int m, int n, vector<vector<int>>& indices) {
        int r,c;
          int count = 0;
          //step 1 = init array with 0's
          vector<vector<int>> mat(m,vector<int>(n,0));

          //step 2 take r of i and c of i to iterate and increament
          for(int i = 0;i<indices.size();i++){
               r = indices[i][0];
               c = indices[i][1];

                   //step 3 = increament by 1 rows
          for(int j = 0; j< n;j++){
               mat[r][j] += 1;
          }
          

          //step 4 = increament by 1 colomns
          for(int k = 0;k < m;k++){
               mat[k][c] += 1;
          }
          }

      

          //step 5 = count the odd numbers
          for(int i = 0;i<m;i++){
               for(int j = 0; j<n;j++){
                    cout<<mat[i][j];
                    if(mat[i][j] % 2 == 1){
                         count++;
                    }
               }
          }

          return count;

    }
};

File: 1378-CellsWithOddValuesInAMatrix/1378-CellsWithOddValuesInAMatrix.py
# Last updated: 18/10/2025, 09:03:08
class Solution(object):
    def oddCells(self, m, n, indices):
        """
        :type m: int
        :type n: int
        :type indices: List[List[int]]
        :rtype: int
        """
        # Initialize a matrix of m rows and n columns with all 0s
        matrix = [[0 for _ in range(n)] for _ in range(m)]

        # Apply the increment operations
        for r, c in indices:
            # Increment the entire row `r`
            for col in range(n):
                matrix[r][col] += 1
            # Increment the entire column `c`
            for row in range(m):
                matrix[row][c] += 1

        # Count the number of cells with odd values
        count = 0
        for i in range(m):
            for j in range(n):
                if matrix[i][j] % 2 == 1:
                    count += 1

        return count
